examples/animation/locomotion.html

```diff
@@ -0,0 +1,431 @@
+            var controlPanel = new pcui.Panel({
+                headerText: 'Controls',
+                collapsible: true
+            });
+            controlPanel.content.dom.setAttribute('style', 'display: flex; align-items: center; flex-wrap: wrap;')
+            controlPanel.collapsed = false;
+            var jumpButton = new pcui.Button({
+                text: 'Jump'
+            });
+            jumpButton.dom.setAttribute('style', 'width: 100%;')
+            jumpButton.on('click', function() {
+                var isJumping = characterEntity.anim.baseLayer.activeState === 'Jump';
+                if (!isJumping) {
+                    this.jumpTime = 0;
+                    characterEntity.anim.setTrigger('jump');
+                }
+            });
+            controlPanel.append(jumpButton);
+            var jogToggleLabel = new pcui.Label({
+                text: 'Run:'
+            });
+            controlPanel.append(jogToggleLabel);
+            var jogToggle = new pcui.BooleanInput({
+                type: 'toggle'
+            });
+            jogToggle.on('change', function(value) {
+                if (value) {
+                    characterEntity.anim.setInteger('speed', 2);
+                } else {
+                    characterEntity.anim.setInteger('speed', 1);
+                }
+            });
+            controlPanel.append(jogToggle);
+
+            controlPanel.dom.setAttribute('style', `position: fixed; right: 24px; top: 24px; width: ${width}px; z-index: 9999;`);
+
+            document.body.appendChild(controlPanel.dom);
+
+
+            var keyboard = new pc.Keyboard(window);
+
+            var Locomotion = pc.createScript('Locomotion');
+
+            var characterDirection = new pc.Vec3(1, 0, 0);
+            var targetPosition = new pc.Vec3(2, 0, 2);
+
+            // initialize code called once per entity
+            Locomotion.prototype.initialize = function() {
+                // app.mouse.disableContextMenu();
+                // app.mouse.on(pc.EVENT_MOUSEDOWN, this.onMouseDown, this);
+                document.addEventListener("mousedown", this.onMouseDown);
+            };
+
+            Locomotion.prototype.onMouseDown = function (event) {
+                if (event.button !== 0) return;
+                // Set the character target position to a position on the plane that the user has clicked
+                var cameraEntity = app.root.findByName('Camera');
+                var near = cameraEntity.camera.screenToWorld(event.x, event.y, 0.1);
+                var far = cameraEntity.camera.screenToWorld(event.x, event.y, 1000.0);
+                var result = app.systems.rigidbody.raycastFirst(far, near);
+                if (result) {
+                    targetPosition = new pc.Vec3(result.point.x, 0, result.point.z);
+                    characterEntity.anim.setInteger('speed', jogToggle.value ? 2 : 1);
+                }
+            };
+
+            function speedForState(state) {
```

Comment:
comments please

Reply:
For some reason it's not showing up but i've added this in.